src/diffusers/models/cross_attention_processors.py

```diff
@@ -0,0 +1,148 @@
+# Copyright 2022 The HuggingFace Team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import torch
+
+from ..utils.import_utils import is_xformers_available
+
+
+if is_xformers_available():
+    import xformers
+    import xformers.ops
+else:
+    xformers = None
+
+
+class CrossAttentionProcMixin:
+
+    def __init__(self, head_size, upcast_attention):
+        self.head_size = head_size
+        self.upcast_attention = upcast_attention
+
+    def __call__(self, hidden_states, query_proj, key_proj, value_proj, context=None):
+        raise NotImplementedError("Make sure this method is overwritten in the subclass.")
+
+    def batch_to_head_dim(self, tensor, head_size):
+        batch_size, seq_len, dim = tensor.shape
+        tensor = tensor.reshape(batch_size // self.head_size, self.head_size, seq_len, dim)
+        tensor = tensor.permute(0, 2, 1, 3).reshape(batch_size // self.head_size, seq_len, dim * self.head_size)
+        return tensor
+
+    def head_to_batch_dim(self, tensor, head_size):
+        batch_size, seq_len, dim = tensor.shape
+        tensor = tensor.reshape(batch_size, seq_len, self.head_size, dim // self.head_size)
+        tensor = tensor.permute(0, 2, 1, 3).reshape(batch_size * self.head_size, seq_len, dim // self.head_size)
+        return tensor
+
+    def get_attention_scores(self, query, key):
+        dtype = query.dtype
+        if self.upcast_attention:
+            query = query.float()
+            key = key.float()
+
+        attention_scores = torch.baddbmm(
+            torch.empty(query.shape[0], query.shape[1], key.shape[1], dtype=query.dtype, device=query.device),
+            query,
+            key.transpose(-1, -2),
+            beta=0,
+            alpha=self.scale,
+        )
+        attention_probs = attention_scores.softmax(dim=-1)
+        attention_probs = attention_probs.to(dtype)
+
+        return attention_probs
+
+
+class CrossAttentionProc(CrossAttentionProcMixin):
+
+    def __call__(self, hidden_states, query_proj, key_proj, value_proj, context=None):
```

Comment:
pass all weigths - note in order to support LORA we need to add two more layers (proj and dropout)